### PR TITLE
Option groups

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Infrastructure "Option groups"

#### Purpose of change
Implement option groups for convenience.

#### Describe the solution
Partially ported https://github.com/cataclysmbnteam/Cataclysm-BN/pull/562 by @olanti-p. I didn't touch any change to logging system from that PR, only taken the "option group" infrastructure stuff.
As a proof of concept (and as it was in that PR), put font settings into an option group.

#### Describe alternatives you've considered
None.

#### Testing
Opened `Settings`->`Options` menu, expanded and collapsed font settings option group. Also checked that everything else (prerequisites, mouse selecting and scrolling, changing options and tabs) is working as it should.

#### Additional context
изображение

изображение

### CODE REVIEW

File: src/options.cpp [hunk 1]
```diff
@@ -558,6 +552,62 @@ void options_manager::add( const std::string &sNameIn, const std::string &sPageI
...
+    static Group null_group;
+    if( id.empty() ) {
+        return null_group;
+    }
+    for( const Group &g : groups_ ) {
+        if( g.id_ == id ) {
+            return g;
+        }
+    }
+    debugmsg( "Option group with id '%d' does not exist.", id );
```
Comment (olanti-p) [inline src/options.cpp:607]: Just noticed, there was a type mismatch in this format string in the original PR
```suggestion
    debugmsg( "Option group with id '%s' does not exist.", id );
```

**Comment by ZhilkinSerg** [comment]
Will the group expand on touch event?

**Comment by Night-Pryanik** [comment]
I'm not sure, I can't test it on any touch-able hardware.

**Comment by Night-Pryanik** [comment]
> Will the group expand on touch event?

Ok, tested it at last. The group expands when it's selected and user taps anywhere on the screen.

**Pull request merged**